src/store/index.ts

```diff
@@ -0,0 +1,160 @@
+    state.showDeleteAlert = value;
+  },
+  SET_SHOW_CREATE_ALERT(state: AppState, value: boolean) {
+    state.showCreateAlert = value;
+  },
+};
+const actions = {
+  async fetchPosts({ commit }: { commit: Function }) {
+    try {
+      const response = await axios.get<Post[]>(`${BACKEND_URL}/posts`);
+      console.log('Posts:', response.data);
+      commit('SET_POSTS', response.data);
+      return response.data;
+    } catch (error) {
+      console.error('Error fetching posts:', error);
+      throw error;
+    }
+  },
+  async fetchPost({ commit }: { commit: Function }, postId: string) {
+    try {
+      const response = await axios.get(`${BACKEND_URL}/posts/${postId}`);
+      commit('SET_SELECTED_POST', response.data);
+      commit('SET_SELECTED_TAGS', response.data.tags);
+    } catch (error) {
+      console.error('Error fetching post:', error);
+    }
+  },
+  async updatePost({ state }: { state: AppState }, postId: number) {
+    const updatedPost: Post = {
+      id: postId,
+      image: state.selectedPost.image,
+      tags: state.selectedPost.tags,
+      description: state.selectedPost.description,
+      created_at: state.selectedPost.created_at,
+      updated_at: new Date().toISOString(),
+    };
+
+    try {
+      const response = await axios.put(`${BACKEND_URL}/posts/${postId}`, updatedPost);
+      console.log('Post updated:', response.data);
+      router.push('/post-list');
+    } catch (error) {
+      console.error('Error updating post:', error);
+    }
+  },
+  async deletePost({ commit }: { commit: Function }, postId: string) {
+    try {
+      await axios.delete(`${BACKEND_URL}/posts/${postId}`);
+      console.log('Post deleted successfully');
+      commit('REMOVE_DELETED_POST', postId);
+      commit('SET_SHOW_DELETE_ALERT', true);
+      setTimeout(() => {
+        commit('SET_SHOW_DELETE_ALERT', false);
+      }, 3000);
+    } catch (error) {
+      console.error('Error deleting post:', error);
+      throw error;
+    }
+  },
+  async savePost({ commit }: { commit: Function }, post: Post) {
+    const newPost: Post = {
+      ...post,
+      created_at: new Date().toISOString(),
+      updated_at: new Date().toISOString(),
+    };
+
+    try {
+      const response = await axios.post(`${BACKEND_URL}/posts`, newPost);
+      console.log('Post created:', response.data);
+      router.push('/post-list');
+      commit('SET_SHOW_CREATE_ALERT', true);
+      setTimeout(() => {
+        commit('SET_SHOW_CREATE_ALERT', false);
+      }, 3000);
+    } catch (error) {
+      console.error('Error creating post:', error);
+      throw error;
+    }
+  },
+};
+const getters = {
+  sortedPosts(state: AppState) {
+    return sortedPosts.value;
+  },
+  showDeleteAlert(state: AppState) {
+    return state.showDeleteAlert;
+  },
+  showCreateAlert(state: AppState) {
+    return state.showCreateAlert;
+  },
+};
+export default createStore({
+  state,
+  mutations,
+  actions,
+  getters,
+<<<<<<< HEAD
```

Comment:
I get this `error  Parsing error: Merge conflict marker encountered` when running your project. You should resolve the conflict and remove lines of the form:
`<<<<<<< HEAD`, `=======`, `>>>>>>> c9dab5ca18d9dbda14c1380fa4f34fef4636e09a`